examples/erc20/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+#[macro_use]
+extern crate serde; // Provides `Serialize` and `Deserialize`.
+
+use mantle::{Address, Context, Event, Service};
+
+use std::collections::{hash_map::Entry, HashMap, HashSet};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Eq, PartialEq, failure::Fail)]
+pub enum Error {
+    #[fail(display = "Unknown error occured.")]
+    Unknown,
+
+    #[fail(display = "Only existing admins can perform this operation.")]
+    AdminPrivilegesRequired,
+
+    #[fail(display = "Insuffient funds for transfer from {:?}.", address)]
+    InsufficientFunds { address: Address },
+
+    #[fail(display = "Address {:?} has no allowance from address {:?}.", from, to)]
+    NoAllowanceGiven { from: Address, to: Address },
+
+    #[fail(
+        display = "Transfer request {} exceeds allowance {}.",
+        amount, allowance
+    )]
+    RequestExceedsAllowance { amount: u64, allowance: u64 },
+}
+
+#[derive(Service, Default)]
+pub struct ERC20Token {
+    /// Total supply
```

Comment:
imho these one-liner comments that say the field name are a bit noisy. I'd just remove them; the names are clear enough.

Reply:
Done.